src/streamer.rs

```diff
@@ -205,6 +205,11 @@ fn find_next_missing(
     Ok(reqs)
 }
 
+pub fn calculate_repair_window(crdt: &Arc<RwLock<Crdt>>, consumed: u64, received: u64) -> u64 {
```

Comment:
This function doesn't need to be `pub` and doesn't require any information from `crdt` besides its table length. I'd suggest passing in `peers: i64` instead. That'll make your test ~90% shorter and just as meaningful. Also, still looking for detailed documentation above this new `calculate_repair_window` function explaining why it needs to exist. It's possible that only a more precise function name is needed. "calculate repair window" means almost nothing at all to me. Maybe `calculate_repair_window_index_upper_bound`? I really don't understand why `received` might morph into `consumed` or `0` here, and it sounds like it's quite important! I'd appreciate your help here in clarifying this nuance.